src/perf_tools/scheduler_trace.cr

```diff
@@ -0,0 +1,134 @@
+require "./common"
+
+module PerfTools::SchedulerTrace
+  {% if flag?(:unix) %}
+    # Installs a signal handler to call `.print_runtime_status` on demand.
+    #
+    # Uses `SIGUSR1` by default but you may configure another signal, for
+    # example `LibC::SIGRTMIN + 7`. You may also register multiple signals, one
+    # with fiber detail and the another without for example.
+    def self.on(signal : Int32 = Signal::USR1.value, details = false) : Nil
+      # not using Signal#trap so the signal will be handled directly instead
+      # of through the event loop that may have to wait (or be blocked in
+      # the worst case):
+      action = LibC::Sigaction.new
+      action.sa_flags = LibC::SA_RESTART
+      action.sa_sigaction = LibC::SigactionHandlerT.new do |_, _, _|
+        print_runtime_status(details)
```

Comment:
This line doesn't compile: we can't pass a variable (here `details`) to a C closure _and_ we can't pass custom data to the sigaction handler either.

We either need a global variable (meh) or keep a `Hash(Int32, Bool)` as a class property (sounds better).

Reply:
Since we only need to cover two cases, maybe we could create two different procs?
```cr
if details 
  LibC::SigactionHandlerT.new do |_, _, _|
    print_runtime_status(true)
  end
else
  LibC::SigactionHandlerT.new do |_, _, _|
    print_runtime_status(false)
  end
end
```